jobs/admin.py

```diff
@@ -99,127 +99,134 @@ def send_status_update_meetup(modeladmin, request, queryset):
-                self.admin_site.admin_view(self.restore_job),
-                name='restore_job'
+                self.admin_site.admin_view(self.restore),
+                name='restore_%s' % self.get_print_name()
             ),
             url(
                 r'^(?P<id>\d+)/publish/$',
-                self.admin_site.admin_view(self.publish_job),
-                name='publish_job'
+                self.admin_site.admin_view(self.publish),
+                name='publish_%s' % self.get_print_name()
             ),
         )
         return my_urls + urls
 
-    def assign_job_reviewer(self, request, id):
-        job = get_object_or_404(Job, id=id)
-        job.assign(request.user)
+    def get_print_name(self):
+        return self.model._meta.model_name
+
+    def not_expired(self, obj):
+        return obj.not_expired
+    not_expired.boolean = True
+    not_expired.admin_order_field = 'not_expired'
+
+    def assign_reviewer(self, request, id):
+        post = get_object_or_404(self.model, id=id)
+        post.assign(request.user)
         messages.add_message(
             request,
             messages.INFO,
-            '{0} is now assigned.'.format(job)
+            '{0} is now assigned.'.format(post)
         )
-        return redirect('/admin/jobs/job/%s/' % id)
+        return redirect('/admin/jobs/%s/%s/' % (self.get_print_name(), id))
 
-    def unassign_job_reviewer(self, request, id):
-        job = get_object_or_404(Job, id=id)
-        job.unassign()
+    def unassign_reviewer(self, request, id):
+        post = get_object_or_404(self.model, id=id)
+        post.unassign()
         messages.add_message(
             request,
             messages.INFO,
-            '{0} is now unassigned.'.format(job)
+            '{0} is now unassigned.'.format(post)
         )
-        return redirect('/admin/jobs/job/%s/' % id)
+        return redirect('/admin/jobs/%s/%s/' % (self.get_print_name(), id))
 
-    def accept_job(self, request, id):
-        job = get_object_or_404(Job, id=id)
-        job.accept()
+    def accept(self, request, id):
+        post = get_object_or_404(self.model, id=id)
+        post.accept()
         messages.add_message(
             request,
             messages.INFO,
-            '{0} is now accepted.'.format(job)
+            '{0} is now accepted.'.format(post)
         )
-        return redirect('/admin/jobs/job/%s/' % id)
+        return redirect('/admin/jobs/%s/%s/' % (self.get_print_name(), id))
 
-    def reject_job(self, request, id):
-        job = get_object_or_404(Job, id=id)
-        job.reject(option='job offer')
+    def reject(self, request, id):
+        post = get_object_or_404(self.model, id=id)
+        post.reject()
         messages.add_message(
             request,
             messages.INFO,
-            '{0} is now rejected - an email to submitter was sent.'.format(job)
+            '{0} is now rejected - an email to submitter was sent.'.format(post)
```

Comment:
Does this message get shown even if there was an error sending the email?  (Maybe just something to keep in mind - only pointing it out because we have a check in the admin send_status_update..() methods in case of an error sending the mail.
Also, you may want to either put a try/except around the send_mail calls and catch the error if the email sending fails, OR set ```fail_silently=True``` and use the if statement used in the in the admin send_statu_update() methods.  I think at the moment that if the email fails the app will throw an error (because fail_silently is false by default), which will look to the user like the whole action has failed, rather than just the mail.